rust/operator-binary/src/controller.rs

```diff
@@ -427,7 +428,7 @@ pub async fn reconcile_nifi(nifi: Arc<NifiCluster>, ctx: Arc<Ctx>) -> Result<Act
                 CONTROLLER_NAME,
                 nifi.deref(),
                 &NifiStatus {
-                    deployed_version: Some(nifi_product_version.to_string()),
+                    deployed_version: Some(resolved_product_image.product_version),
```

Comment:
A change in the product image that bumps the stackable version can be breaking. Not entirely sure if we should use `resolved_product_image.product_version` or `resolved_product_image.app_version_label`.
Anyways: A comment - regardless of the outcome - what is considered as a trigger for a version_change would be great.
CC @soenkeliebau

Reply:
The trigger is the nifi version without stackable version as far as i know. This is just replacing the existing logic one to one?

Reply:
Yes, this replicates existing logic